sklearn/pipeline.py

```diff
@@ -657,6 +659,16 @@ def n_features_in_(self):
         # delegate to first step (which will call _check_is_fitted)
         return self.steps[0][1].n_features_in_
 
+    @property
+    def __sk_is_fitted__(self):
+        """Indicate whether pipeline has been fit."""
+        try:
+            # check if the last step of the pipeline is fitted
+            check_is_fitted(self.steps[-1][1])
```

Comment:
The fact that we only check the last step of the pipeline is to be nice with users that have a currently working pipeline with their own custom stateless transformers that would fail the `check_is_fitted` check if we were to have this property call this `check_is_fitted` on all steps?

Or is there another reason?

Reply:
@glemaitre 's script was checking for the first step, I thought it makes sense to do it for the last step. I'm agnostic on how we do it. I'll add a comment.